shared/AppInsightsCommon/src/Util.ts

```diff
@@ -652,16 +658,35 @@ export class Util {
 
 export class UrlHelper {
     private static document: any = getDocument()||{};
-    private static htmlAnchorElement: HTMLAnchorElement;
+
+    private static _htmlAnchorIdx: number = 0;
+    // Use an array of temporary values as it's possible for multiple calls to parseUrl() will be called with different URLs
+    // Using a cache size of 5 for now as it current depth usage is at least 2, so adding a minor buffer to handle future updates
+    private static _htmlAnchorElement: HTMLAnchorElement[] = [null, null, null, null, null];
 
     public static parseUrl(url: string): HTMLAnchorElement {
-        if (!UrlHelper.htmlAnchorElement) {
-            UrlHelper.htmlAnchorElement = !!UrlHelper.document.createElement ? UrlHelper.document.createElement('a') : { host: UrlHelper.parseHost(url) }; // fill host field in the fallback case as that is the only externally required field from this fn
```

Comment:
There was a test error with just using the single instance which caused the tests to occasionally fail -- as the used value kept getting overwritten.